BookingService/BookingService.Testing/UserServiceTests.cs

```diff
@@ -0,0 +1,315 @@
+﻿using AutoMapper;
+using BookingService.Application;
+using BookingService.Application.Abstract;
+using BookingService.Application.DTOs;
+using BookingService.Application.Validation;
+using BookingService.Domain;
+using BookingService.Domain.Constants;
+using BookingService.Domain.Models;
+using FluentAssertions;
+using Microsoft.AspNetCore.Identity;
+using Moq;
+
+namespace BookingService.Testing
+{
+    public class UserServiceTests
+    {
+        private readonly Mock<ITokenService> _mockTokenService;
+        private readonly Mock<ICustomUserManager> _mockCustomUserManager;
+        private readonly Mock<ICustomRoleManager> _mockCustomRoleManager;
+        private readonly Mock<IMapper> _mockMapper;
+        private readonly UserService _userService;
+        private readonly LoginDtoValidator _loginDtoValidator;
+        private readonly RegisterDtoValidator _registerDtoValidator;
+
+        public UserServiceTests()
+        {
+            _mockTokenService = new Mock<ITokenService>();
+            _mockCustomUserManager = new Mock<ICustomUserManager>();
+            _mockCustomRoleManager = new Mock<ICustomRoleManager>();
+            _mockMapper = new Mock<IMapper>();
+
+            _userService = new UserService(_mockTokenService.Object, _mockCustomUserManager.Object, _mockCustomRoleManager.Object, _mockMapper.Object);
+
+            _loginDtoValidator = new LoginDtoValidator();
+            _registerDtoValidator = new RegisterDtoValidator();
+        }
+
+        [Fact]
+        public async Task LoginAsync_ShouldReturnJwtToken_WhenEmailAndPasswordAreValid()

```

Comment:
Remove AAA comments bc it's obvious for programmers which section of the test relates to which section of AAA